spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/consumerAuthenticationProvider/apiKeyProvider/ApiKeyProvider.java

```diff
@@ -0,0 +1,53 @@
+package eu.coatrack.proxy.security.consumerAuthenticationProvider.apiKeyProvider;
+
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.api.ApiKey;
+import eu.coatrack.proxy.security.consumerAuthenticationProvider.apiKeyProvider.apiKeyFetcher.ApiKeyFetcher;
+import eu.coatrack.proxy.security.consumerAuthenticationProvider.apiKeyProvider.localApiKeyManager.LocalApiKeyManager;
+import eu.coatrack.proxy.security.exceptions.ApiKeyFetchingFailedException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import static org.apache.commons.codec.digest.DigestUtils.sha256Hex;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class ApiKeyProvider {
+
+    private final ApiKeyFetcher apiKeyFetcher;
+    private final LocalApiKeyManager localApiKeyManager;
+
+    public ApiKey getApiKeyEntityByApiKeyValue(String apiKeyValue) {
```

Comment:
Does this still make sense after introducing hashing? If yes, we should adjust the names to make more clear where the hashed values are used and where not.

Reply:
Originally we needed the API key entity to create the Authentication object:

```
String serviceUriIdentifier = apiKey.getServiceApi().getUriIdentifier();
        ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKey.getKeyValue(), Collections.singleton(
                new SimpleGrantedAuthority(ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX
                        + serviceUriIdentifier)));
        apiKeyAuthToken.setAuthenticated(true);
```

I think this information is required for the post filters and the subsequently called metrics logic. I think that we can do without API key entities, when [#149](https://github.com/coatrack/coatrack/issues/149) is solved.